### PR TITLE
Updates panel for cancelled notes + small clean up

### PR DESCRIPTION
**Pull request by orelbn**
*Issue #:* /bcgov/entity#16926

*Description of changes:*
- Updates the unit note panel header to be (Cancelled) when unit note status is cancelled
- Overrides unit note information in panel using that of the cancel note information
- Moves panel logic to its own composable.
- Creates UnitNotePanel interface to allow for adding UI local properties specific to rendering the panel.
- Updates and adds tests

Note for reviewers:

You can see the applied changes in MHR #107714

Cancel panel

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

**Comment by pwei1018** [comment]
Temporary Url for review: https://bcregistry-assets-dev--pr-1511-egpvn5t1.web.app

File: ppr-ui/src/composables/mhrInformation/useMhrUnitNotePanel.ts [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+   */
+
+  // Identify if a unit note is notice of caution or an extended/continued notice of caution
+  const isNoticeOfCautionOrRelatedDocType = (note: UnitNoteIF): boolean => {
+    return [UnitNoteDocTypes.NOTICE_OF_CAUTION,
+      UnitNoteDocTypes.CONTINUED_NOTE_OF_CAUTION,
+      UnitNoteDocTypes.EXTENSION_TO_NOTICE_OF_CAUTION].includes(note.documentType)
+  }
+
+  const handleCancelledUnitNotes = (unitNotes: UnitNoteIF[]): UnitNotePanelIF[] => {
```
Comment (dimak1) [inline ppr-ui/src/composables/mhrInformation/useMhrUnitNotePanel.ts:22]: We usually 'handle...' an action from the emitted event, but not a biggie.

File: ppr-ui/src/composables/mhrInformation/useMhrUnitNotePanel.ts [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+    // Since notes are in order from most recent creation date and time,
+    // the cancel notes should be added to map before the corresponding note is encountered
+    unitNotes.forEach((note: UnitNotePanelIF) => {
+      if (note.documentType === UnitNoteDocTypes.NOTE_CANCELLATION) {
+        cancelUnitNotesMap.set((note as CancelUnitNoteIF).cancelledDocumentRegistrationNumber, note as CancelUnitNoteIF)
+      } else if (cancelUnitNotesMap.has(note.documentRegistrationNumber)) {
+        // Adds cancelledDateTime and overrides remarks and givingNoticeParty
+        const cancelNote = cancelUnitNotesMap.get(note.documentRegistrationNumber)
+        note.remarks = cancelNote.remarks
```
Comment (dimak1) [inline ppr-ui/src/composables/mhrInformation/useMhrUnitNotePanel.ts:34]: Here not sure when remarks are replaced. Your PR description says it does overwrite the unit note, but I dont have the full context.

**Review by dimak1 — APPROVED** [review]
Looks great! No actionable comments.

**Review by doug-lovett — APPROVED** [review]
Very nice.

File: ppr-ui/src/composables/mhrInformation/useMhrUnitNotePanel.ts (hunk 2 above)
Comment (orelbn) [reply ppr-ui/src/composables/mhrInformation/useMhrUnitNotePanel.ts:34]: The cancel note filling has a editable remarks field that is prefilled with the note being cancelled remarks, what is in that field is shown in the panel as the note's remarks when the note is cancelled.

image
image

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://bcregistry-assets-dev--pr-1511-egpvn5t1.web.app

File: ppr-ui/src/composables/mhrInformation/useMhrUnitNotePanel.ts (hunk 2 above)
Comment (dimak1) [reply ppr-ui/src/composables/mhrInformation/useMhrUnitNotePanel.ts:34]: Thanks for clarification.

**Pull request merged**